sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala

```diff
@@ -985,4 +976,59 @@ class CachedTableSuite extends QueryTest with SQLTestUtils with SharedSQLContext
     val queryStats3 = query().queryExecution.optimizedPlan.stats.attributeStats
     assert(queryStats3.map(_._1.name).toSet === Set("c0", "v1", "v2"))
   }
+
+  test("Refresh Qualified Tables") {
+    withTempDatabase { db =>
+      withTempPath { path =>
+        spark.catalog.createTable(
+          s"$db.cachedTable",
+          "PARQUET",
+          StructType(Array(StructField("key", StringType))),
+          Map("LOCATION" -> path.toURI.toString))
+        withCache(s"$db.cachedTable") {
+          spark.catalog.cacheTable(s"$db.cachedTable", MEMORY_ONLY)
+          assertCached(spark.table(s"$db.cachedTable"), s"$db.cachedTable", MEMORY_ONLY)
+          assert(spark.catalog.isCached(s"$db.cachedTable"),
+            s"Table '$db.cachedTable' should be cached")
+
+          spark.catalog.refreshTable(s"$db.cachedTable")
+          assertCached(spark.table(s"$db.cachedTable"), s"$db.cachedTable", MEMORY_ONLY)
+          assert(spark.catalog.isCached(s"$db.cachedTable"),
+            s"Table '$db.cachedTable' should be cached after refresh")
+
+          activateDatabase(db) {
+            assertCached(spark.table("cachedTable"), s"$db.cachedTable", MEMORY_ONLY)
+            assert(spark.catalog.isCached("cachedTable"),
+              "Table 'cachedTable' should be cached after refresh")
```

Comment:
Ur, this seems to be a copy&paste typo. Did you want `s"Table '$db.cachedTable' should be cached")` here?
Also, it looks like you may want to uncache before `activateDatabase(db)` at line 998 and do `cacheTable` at line 1000.

Reply:
I would like to keep the cache without uncaching it on line 1000. It is because i would like to make sure refreshing the table again with unqualified table name would not change the cache name.  

I have refactor the test a bit with some more explanations. I hope it looks better now.